### ISSUE TITLE
TiKV gRPC server coredump when TLS is enabled

### ISSUE THREAD

**Issue by overvenus** [tikv/tikv#7209]
## Bug Report

TiKV core dumps in BR TLS tests, see more https://github.com/pingcap/br/pull/202

### What version of TiKV are you using?

```
./tikv-server -V
TiKV
Release Version:   4.1.0-alpha
Git Commit Hash:   e364c57f69151c7fb4b00de591217f9edfedd61b
Git Commit Branch: master
UTC Build Time:    2020-03-21 07:37:31
Rust Version:      rustc 1.42.0-nightly (0de96d37f 2019-12-19)
Enable Features:   jemalloc portable sse protobuf-codec
Profile:           dist_release
```

### Steps to reproduce

Run BR integration tests

### What did you expect?

TiKV should start normally or report an error.

### What did happened?

Core dumped.

**Comment by hunterlxt** [comment]
Please add detailed reproduction steps here, just `make integration_test` seems to fail, this script fails any tests

**Comment by july2993** [comment]
[TiDB Binlog](https://github.com/pingcap/tidb-binlog) meets this too


reproduction steps:
Int the tidb-binlog repo:
1. put  tidb/tikv/pd binary in bin/
2. run `make`
3. run `sh tests/run.sh --debug`


if normally you can see:
```
Verifying drainer is started...
{"PumpPos":{"pump:8250":415530729860235266},"Synced":false,"LastTS":415530729860235266,"TsMap":""}
You may now debug from another terminal. Press [ENTER] to continue.
```

but the tikv that enable TLS Core dumped

you can check the log in `/tmp/tidb_binlog_test/`

the upstream tikv's log end at:
```
[2020/03/25 15:49:52.692 +08:00] [INFO] [<unknown>] ["New connected subchannel at 0x7fd05b879020 for subchannel 0x7fd00ca67380"]
[2020/03/25 15:49:52.693 +08:00] [INFO] [util.rs:427] ["connecting to PD endpoint"] [endpoints=https://127.0.0.1:2379]
[2020/03/25 15:49:52.696 +08:00] [INFO] [<unknown>] ["New connected subchannel at 0x7fd05b878240 for subchannel 0x7fd00ca67380"]
[2020/03/25 15:49:52.697 +08:00] [INFO] [util.rs:486] ["connected to PD leader"] [endpoints=https://127.0.0.1:2379]
[2020/03/25 15:49:52.697 +08:00] [INFO] [util.rs:204] ["heartbeat sender and receiver are stale, refreshing ..."]
[2020/03/25 15:49:52.697 +08:00] [WARN] [util.rs:223] ["updating PD client done"] [spend=8.451843ms]
[2020/03/25 15:49:57.243 +08:00] [ERROR] [<unknown>] ["assertion failed: prior > 0"]
[2020/03/25 15:49:57.243 +08:00] [ERROR] [<unknown>] ["assertion failed: prior > 0"]
```

**Comment by WangXiangUSTC** [comment]
dose this issue fixed in release-4.0?

**Comment by hunterlxt** [comment]
> dose this issue fixed in release-4.0?

No

**Comment by WangXiangUSTC** [comment]
release 4.0 binlog's ci test is still failed because of this issue, can we cherry pick the fix to 4.0?
